torch/fx/examples/vmap.py

```diff
@@ -0,0 +1,177 @@
+# This example is provided only for explanatory and educational purposes.
+import torch
+import torch.nn as nn
+import torch.fx as fx
+from torch.fx import Proxy
+
+import torch.fx
+from torch.fx.passes.shape_prop import ShapeProp
+
+# Batching Rules
+# ---------------
+# Giving a full overview of how vmap is implemented is outside the scope of
+# this example, but we will provide a brief description of batching rules. The
```

Comment:
Is there an external reference you can link to if the user is interested in learning more?

Reply:
Unfortunately, not really.